### PR TITLE
add CustomZombieSpawnManager

### PR DESCRIPTION
**Pull request by HSGamer**
* I want to spawn my own zombies

### CODE REVIEW

File: src/main/java/plugily/projects/villagedefense/arena/Arena.java [hunk 1]
```diff
@@ -578,7 +578,7 @@ public Main getPlugin() {
     return plugin;
   }
 
-  protected void addZombie(Zombie zombie) {
+  public void addZombie(Zombie zombie) {
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/arena/Arena.java:581]: Revert this and use `getZombies` which returns the `zombies` modifiable list.

File: src/main/java/plugily/projects/villagedefense/arena/managers/ZombieSpawnManager.java [hunk 2]
```diff
@@ -39,6 +39,26 @@
  */
 public class ZombieSpawnManager {
 
+  private static final List<CustomZombieSpawnManager> CUSTOM_ZOMBIE_SPAWN_MANAGERS = new ArrayList<>();
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/arena/managers/ZombieSpawnManager.java:42]: Remove `static`

File: src/main/java/plugily/projects/villagedefense/arena/managers/CustomZombieSpawnManager.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package plugily.projects.villagedefense.arena.managers;
+
+import plugily.projects.villagedefense.arena.Arena;
+
+import java.util.Random;
+
+public interface CustomZombieSpawnManager {
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/arena/managers/CustomZombieSpawnManager.java:7]: Document that this is only used for API purposes.

File: src/main/java/plugily/projects/villagedefense/arena/managers/ZombieSpawnManager.java (hunk 2 above)
Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/arena/managers/ZombieSpawnManager.java:42]: Why?

**Comment by montlikadani** [comment]
In fact, I don’t think we would need this very much, as not many people use the VD API, you will use this implementation for now, but who will know in the future that it also exists? I mean these “half-beginners” who would like to summon unique zombies but don’t know how. This issue is likely to emerge after 2 years (possibly). One of us would document this on our wiki page, but even then only one person will be interested after a minimum of half a year. Instead of this current implementation, we would create custom configurable zombies, but that would be a big effort and a lot of time as well as big and complicated to implement.

File: src/main/java/plugily/projects/villagedefense/arena/managers/ZombieSpawnManager.java [hunk 4]
```diff
@@ -39,6 +39,26 @@
...
+  }
+
+  /**
+   * Remove a custom zombie spawn manager
+   *
+   * @param customZombieSpawnManager the manager
+   */
+  public static void removeCustomZombieManager(CustomZombieSpawnManager customZombieSpawnManager) {
+    CUSTOM_ZOMBIE_SPAWN_MANAGERS.remove(customZombieSpawnManager);
+  }
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/arena/managers/ZombieSpawnManager.java:60]: You could remove these two methods and make the `CUSTOM_ZOMBIE_SPAWN_MANAGERS` property `public`.

**Comment by HSGamer** [comment]
I know.
I even have an idea to recode the whole zombie system in my fork.
But for now, this is what I need

**Comment by HSGamer** [comment]
for now, this works like an injector for the zombie spawner.
in the future, you guys may have enough time to implement your configurable zombies.

**Pull request merged**